### ISSUE TITLE
Setting sourceDirectories and classDirectories

### ISSUE THREAD

**Issue by PaishoIroh** [mannodermaus/android-junit5#65]
Hi, I've been able to set up the plugin for my current project but seem to have an issue when I take a look at the Jacoco report. The current Jacoco report doesn't allow me to see highlighted pieces of code to better know which were covered by tests and which I missed. 

Reading the FAQ from Jacoco I noticed it was related to to the sourceDirectories and classDirectories of the project but I can't seem to specify them under the jacocoOptions portion. Does this need to be done in a different area or specified in a different way?

**Comment by mannodermaus** [comment]
The plugin will set up the `sourceDirectories` and `classDirectories` itself when it constructs the Jacoco companion tasks, based on what is configured in the Android variant. So you're saying that the report is generated and you do see the coverage, but you don't have code preview inside of the report? Right now, there's no way to specify additional class and source directories, you can only filter out certain inputs. However, if you see the coverage itself, this is configured properly. I'm curious if there is more configuration for Jacoco so that it enables code preview, too.

**Comment by PaishoIroh** [comment]
Yes the report is generated correctly and coverage is visible but no code preview. I was trying to find out more about the directories since that is the suggestion on the Jacoco FAQ (http://www.jacoco.org/jacoco/trunk/doc/faq.html). Maybe the issue lies with one of these parts?

1. Class files must be compiled with debug information to contain line numbers.
2. Source files must be properly supplied at report generation time. I.e. specified source folders must be the direct parent of the folders that define the Java packages.

Sorry I can't be of much help. I'm barely learning to use this.

**Comment by mannodermaus** [comment]
Thanks for the research! I will experiment with the second point in particular, maybe there is something we can improve on.

**Comment by autonomousapps** [comment]
I came here to report this very issue.
![screenshot from 2018-03-29 21-39-19](https://user-images.githubusercontent.com/5402301/38124905-4c709d48-339a-11e8-963e-2ae3408c00c3.png)
As you can see, you cannot click any method to dive into the code preview. I thought this was because I was looking at a Kotlin source file, but the issue is the same for Java source files. In the past, with JUnit4 tests, I had to manually configure Jacoco tasks, and I was able to overcome this by explicitly specifying the source directories like so:
```
task connectedDebugAndroidTestCoverage(dependsOn: ["connectedDebugAndroidTest"], type: JacocoReport) {
    sourceDirectories = files(["src/main/java", "src/debug/java", "src/flavor/buildtype/java" /* etc */])
}
```

**Comment by mannodermaus** [comment]
Thanks for the input! Are you experiencing this inside a library module, by any chance? Trying to identify if this might be related to #72. The `sourceDirectories` are supposed to be identical between a JUnit task and its Jacoco companion, but looks like there's something amiss here.

**Comment by autonomousapps** [comment]
Nope, this is a `com.android.application` module. It implements an interface defined in a `com.android.library` module, if that matters. Also, FWIW, I have not experienced trouble _running_ tests in library modules. But in all modules, I can't click into code previews in the test reports :man_shrugging:

**Comment by mannodermaus** [comment]
After investigating and experimenting a little more, I found that the directories applied to the plugin's Jacoco tasks aren't correct. When using the auto-generated tasks, this is the output (for a sample project with a single file, `com.example.MyActivity`:

```

[android-junit5]: Assembled Jacoco Code Coverage for JUnit 5 Task 'junitPlatformTestDebug':
[android-junit5]: |__ Execution Data: /home/workspace/MyApplication/app/build/jacoco/junitPlatformTestDebug.exec
[android-junit5]: |__ Source Dirs: /home/workspace/MyApplication/app/src/main/java/com/example/MyActivity.java
[android-junit5]: |__ Class Dirs: /home/workspace/MyApplication/app/build/intermediates/javac/debug/compileDebugJavaWithJavac/classes/com/example/MyActivity.class
```

The directories aren't directories at all, but rather a list of the included files themselves. When I use a custom task on the same execution data, I get the proper code preview:

```groovy
task customJacoco(type: JacocoReport) {
  executionData(file("build/jacoco/junitPlatformTestDebug.exec"))
  sourceDirectories = files("src/main/java")
  classDirectories = files("build/intermediates/javac/debug/compileDebugJavaWithJavac/classes")
}
```

<img width="399" alt="screen shot 2018-05-02 at 12 43 54" src="https://user-images.githubusercontent.com/3897705/39504423-7f759b4a-4e06-11e8-8631-2902512798a0.png">

I'll make sure to have this case handled correctly by the plugin.

**Comment by mannodermaus** [comment]
This will be fixed in the upcoming `1.0.32` release. Until then, please refer to the latest SNAPSHOT version! Thanks again for reporting.
